benches/bin/rpc.rs

```diff
@@ -0,0 +1,355 @@
+            .max_concurrent_stream(DEFAULT_GRPC_CONCURRENT_STREAM)
+            .max_receive_message_len(MAX_GRPC_RECV_MSG_LEN)
+            .max_send_message_len(MAX_GRPC_SEND_MSG_LEN)
+            .build_args();
+
+        let grpc_server = ServerBuilder::new(env.clone())
+            .register_service(create_tikv(h.clone()))
+            .bind("127.0.0.1", 0)
+            .channel_args(channel_args)
+            .build()
+            .unwrap();
+
+        let addr = {
+            let (ref host, port) = grpc_server.bind_addrs()[0];
+            SocketAddr::new(IpAddr::from_str(host).unwrap(), port as u16)
+        };
+
+        BenchTikvServer {
+            server: grpc_server,
+            local_addr: addr,
+            handler: h,
+            rx: None,
+        }
+    }
+
+    pub fn start(&mut self) {
+        self.server.start();
+    }
+
+    // TODO: addrs.
+    pub fn local_addr(&self) -> SocketAddr {
+        self.local_addr
+    }
+
+    pub fn init_recording(&mut self, name: String) -> Result<(), String> {
+        let (tx, rx) = channel();
+        let counter = Arc::new(AtomicUsize::new(0));
+
+        self.rx = Some(rx);
+        self.handler.init_recording(name, counter, tx)
+    }
+
+    pub fn start_recording(&self) -> Result<(), String> {
+        self.handler.start_recording()
+    }
+
+    pub fn stop_recording(&mut self) -> Result<usize, String> {
+        self.rx.take();
+        self.handler.stop_recording()
+    }
+
+    pub fn recv(&self) -> Result<(), String> {
+        if self.rx.is_some() {
+            self.rx.as_ref().unwrap().recv().map_err(|_| "channel has hung up".to_owned())
+        } else {
+            Err("Benchmark is not ready".to_owned())
+        }
+    }
+}
+
+fn new_bench_server() -> (Arc<Environment>, BenchTikvServer) {
+    let env = Arc::new(Environment::new(DEFAULT_GRPC_CONCURRENCY));
+    let mut server = BenchTikvServer::new(env.clone());
+    server.start();
+
+    // Sleep awhile for making sure server is ready.
+    thread::sleep(Duration::from_secs(1));
+
+    (env, server)
+}
+
+// TODO: Stabilize result.
+pub fn bench_raft_rpc() {
+    let name = "raft_client_streaming";
+    printf!("benching RPC on {}\t...", name);
+
+    let (env, mut server) = new_bench_server();
+    let addr = server.local_addr();
+    let mut client = RaftClient::new(env, Config::default());
+
+    let quit = Arc::new(AtomicUsize::new(0));
+    let quit1 = quit.clone();
+    server.init_recording(name.to_owned()).unwrap();
+
+    thread::spawn(move || {
+        // TODO: calc the precise duration for every eventloop tick.
+        let batch_size = 1024;
+        let flush_duration = 1;
+
+        let mut region_id = 0;
+        let sample = RaftMessage::new();
+        loop {
+            if 0 != quit.load(Ordering::Acquire) {
+                return;
+            }
+            region_id += 1;
+            let mut msg = sample.clone();
+            msg.set_region_id(region_id);
```

Comment:
`msg` should contains some payload. And each region may contains several messages.